__tests__/app.test.js

```diff
@@ -161,4 +161,43 @@ describe ('GET /api/articles/:article_id/comments',() => {
         });
     });
   
-})
\ No newline at end of file
+});
+
+//Task 7 POST /api/articles/:article_id/comments
+describe ('POST /api/articles/:article_id/comments',() => {
+    test ('201:should return a new comment when add a new comment by user', () => {
+        const newComment = {username:'butter_bridge',body:'Very Good'};
+        return request(app)
+        .post('/api/articles/1/comments')
+        .send(newComment)
+        .expect(201)
+        .then (({body}) => {
+            const {comments} = body;
+            expect(comments).toHaveProperty("comment_id", expect.any(Number));
+            expect(comments).toHaveProperty("votes", 0);   
+            expect(comments).toHaveProperty("created_at",);   
+            expect(comments.created_at).not.toBeNull();
+            expect(comments).toHaveProperty("author", "butter_bridge");
+            expect(comments).toHaveProperty("body", "Very Good");
+            expect(comments).toHaveProperty("article_id",1);
+        })
+    });
+    test ("404:should return  an error respond when article_id is valid,but does not exist", () => {
+        return request(app)
+            .get("/api/articles/9999/comments")
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("Not found");
+            });
+       });
+
+    test ("400: should return an error if invalid article_id", () => {
+        return request(app)
+            .get("/api/articles/nonsense/comments")
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe("Bad request");
+            });
+    });
+})
+
```

Comment:
The tests you’ve done so far are great, but there’s a few others we need to consider:

- A test that ignores any unnecessary properties if we send a comment with something else as well as a username and body
- An error test for missing required fields, e.g. no username or body properties sent in the comment
- An error test for if the username we send doesn’t exist

Reply:
You mean we do not need to test this one ,right?

A test that ignores any unnecessary properties if we send a comment with something else as well as a username and body

Reply:
Hey Steven, I mean you just need a test if the comment you sent was something like this:
{username: 'butter_bridge' , body: 'Very Good', bananas: 'bananas'}
It should ignore the extra property(bananas) and still post successfully!
Hope this makes sense!

Reply:
thank you. complete your comments